c/eth_addr_reg.c

```diff
@@ -0,0 +1,119 @@
+/**
+ * `ETH Address Registry` layer2 contract
+ * 
+ * This contract introduces two-ways mappings between `eth_address` and
+ * `gw_script_hash`.
+ * 
+ *   - `eth_address` is the address of an Ethereum EOA (externally owned account
+ *     ) or a Polyjuice contract account.
+ * 
+ *   - Godwoken account script hash(a.k.a. `gw_script_hash`) is a key used for
+ *     locating the account lock. Godwoken enforces one-to-one mapping between 
+ *     layer 2 lock script and accountID.
+ */
+
+#include "gw_syscalls.h"
+#include "polyjuice_utils.h"
+#include "sudt_utils.h"
+
+/* MSG_TYPE */
+#define MSG_QUERY_GW_BY_ETH 0
+#define MSG_QUERY_ETH_BY_GW 1
+#define MSG_SET_MAPPING     2
+
+int handle_fee(gw_context_t *ctx, mol_seg_t fee_seg) {
+  if (ctx == NULL) {
+    return GW_FATAL_INVALID_CONTEXT;
+  }
+
+  /* payer's short address */
+  uint8_t payer_account_script_hash[32] = {0};
+  int ret = ctx->sys_get_script_hash_by_account_id(
+      ctx, ctx->transaction_context.from_id, payer_account_script_hash);
+  if (ret != 0) {
+    return ret;
+  }
+  uint64_t short_script_hash_len = DEFAULT_SHORT_SCRIPT_HASH_LEN;
+  /* sudt */
+  mol_seg_t sudt_id_seg = MolReader_Fee_get_sudt_id(&fee_seg);
+  uint32_t sudt_id = *(uint32_t *)sudt_id_seg.ptr;
+  /* amount */
+  mol_seg_t amount_seg = MolReader_Fee_get_amount(&fee_seg);
+  uint128_t amount = *(uint128_t *)amount_seg.ptr;
+  return sudt_pay_fee(ctx, sudt_id, short_script_hash_len,
```

Comment:
This PR https://github.com/nervosnetwork/godwoken-scripts/pull/107 removes the SUDT pay fee feature.
We should do the same here, only support pay CKB as fee.

Reply:
https://github.com/nervosnetwork/godwoken-polyjuice/pull/113/commits/2bedc018e6936d1469413f78c557d492e9407f18